lib/internal/worker.js

```diff
@@ -265,6 +275,13 @@ class Worker extends EventEmitter {
     this[kHandle].startThread();
 
     process.nextTick(() => process.emit('worker', this));
+    // Send current thread loopStart to the worker. In case the loop has not yet
+    // started send it after one iteration.
+    if (milestones[NODE_PERFORMANCE_MILESTONE_LOOP_START] === -1) {
+      setImmediate(() => this[kSendLoopStart]());
```

Comment:
Any reason to use `setImmediate` instead of `process.nextTick` like the line above? The comment seems to indicate we want the next tick, so that seems more appropriate.

Reply:
Yes, we need a full iteration of the loop to complete, at least till the `poll` phase completes. That's what `setImmediate()` is for